### ISSUE TITLE
How to set the time zone of the X axis

### ISSUE THREAD

**Issue by liubin595338764** [tradingview/lightweight-charts#221]
https://jsfiddle.net/TradingView/bmL0vont/

![image](https://user-images.githubusercontent.com/5274428/65504580-e6af9380-def9-11e9-9f8c-158308dba087.png)

How can I set the time zone of the X axis?
My local is the +8 time zone, but it shows the time in the 0 time zone.

`time: 1556877600`  Should show  `2019/05/03 18:00:00`

**Comment by timocov** [comment]
Sorry, but we don't understand Chinese. Please use English.

**Comment by liubin595338764** [comment]
@timocov hi, I have changed to English.

**Comment by timocov** [comment]
> How can I set the time zone of the X axis?

https://github.com/tradingview/lightweight-charts/issues/125#issuecomment-511307557

> And the x-axis time will show an exception on some browsers.

What the browser you have?

**Comment by liubin595338764** [comment]
QQ browser

**Comment by liubin595338764** [comment]
QQ browser : 
`console.log(new Date().toLocaleString())` => `Tue Sep 24 2019 22:46:07 GMT+0800 (CST)`

chrome browser: 
`console.log(new Date().toLocaleString())` => `2019/9/24 下午10:45:48`

**Comment by timocov** [comment]
> QQ browser && HUAWEI mobile phone browser:

Can you please try to set `formatOptions` like we use them here: https://github.com/tradingview/lightweight-charts/blob/b88c8b5844f0aeb171903c8afc784b9f30b49541/src/model/time-scale.ts#L649-L687

It's possible that this browser doesn't support them.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString#Browser_compatibility

**Comment by liubin595338764** [comment]
hi, timocov. I don't know how to use this method, can you give me a list?

**Comment by liubin595338764** [comment]
@timocov 
```html
<div id="time"></div>
<script>
var time = document.querySelector('#time');
time.innerHTML = new Date().toLocaleString('cn', { hour12: false, hour: '2-digit', minute: '2-digit' });
</script>
```
chrome browser:
`13:42`

QQ browser && HUAWEI mobile phone browser:
`Wed Sep 25 2019 13:42:09 GMT+0800 (CST)`

![image](https://user-images.githubusercontent.com/5274428/65572583-1793d500-df9b-11e9-9695-5a7f6d7366c2.png)

**Comment by timocov** [comment]
@liubin595338764 these options are supported even by IE11. I think there is polyfill for that method so you can put it on your page for that browsers.

> The options parameter of toLocaleString is still not supported by some browsers

According to the https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString#Browser_compatibility the `options` doesn't supported in Android WebView only, and I don't think that there is a lot of devices with that webview (afaik in the latest versions of Android (starting from Android 5 I guess) it uses Chrome WebView in common case). I don't why QQ browser doesn't support this parameter so far, but I said - you can try to put polyfills to your page to avoid this issue.

**Comment by liubin595338764** [comment]
@timocov 
About Date's polyfills only `Date.now` and `Date.prototype.toISOString`

**Comment by timocov** [comment]
> QQ browser && HUAWEI mobile phone browser has a very large user volume in China

I don't argue with you, I don't understand why this browser doesn't support new standard yet. IE11 was released a long time ago and it supports `options` argument, the same about Safari 10. We've used some features from es2015 (like Set/Map) and we don't provide any polyfills. If you want to support old browsers or browsers which doesn't support some features, you need to use polyfills.

> About Date's polyfills only Date.now and Date.prototype.toISOString;

What do you mean?

**Comment by liubin595338764** [comment]
>What do you mean?

Date's polyfills only `Date.now` and `Date.prototype.toISOString`;
I haven't found this `toLocaleString` polyfill. Can you help me?

**Comment by timocov** [comment]
Try to look at some intl.js polyfills - might be they help https://github.com/andyearnshaw/Intl.js

**Comment by liubin595338764** [comment]
Intl.js solves the problem of HUAWEI mobile phone browser, but it does not solve the problem of QQ browser.

**Comment by liubin595338764** [comment]
Is there any other way?

**Comment by timocov** [comment]
@liubin595338764 I cannot tell you for sure, but you can try to implement that polyfill yourself. As I said, it's really strange that QQ browser still doesn't support that method and I would suggest to ask developers why they don't implement it yet.

**Comment by liubin595338764** [comment]
thank you so much

**Comment by timocov** [comment]
Hey @liubin595338764, in #386 we've added ability to override default formatter of tick marks, so you can override it to avoid using toLocaleString in your build. It'll be released in 3.0 version.

**Comment by YaoKaiLun** [comment]
I think the problem is here.
<img width="1297" alt="WX20200501-103200@2x" src="https://user-images.githubusercontent.com/11460856/80777228-0ba66c00-8b97-11ea-9cb1-52b33f7e5f30.png">
